pkg/controller/mizar/node_controller.go

```diff
@@ -0,0 +1,307 @@
+(1) https://github.com/h-w-chen/arktos.git, arktos/cmd/arktos-network-controller
+(2) https://github.com/futurewei-cloud/arktos.git, arktos/pkg/controller/nodelifecycle/node_lifecycle_controller
+*/
+
+package mizar
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	coreinformers "k8s.io/client-go/informers/core/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	v1core "k8s.io/client-go/kubernetes/typed/core/v1"
+	corelisters "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+)
+
+const (
+	NODE_KIND          string = "Node"
+	NODE_READY         string = "True"
+	NODE_CREATE        string = "Create"
+	NODE_UPDATE        string = "Update"
+	NODE_DELETE        string = "Delete"
+	NODE_RESUME        string = "Resume"
+	NODE_statusMessage string = "HANDLED"
+)
+
+type NodeController struct {
+	kubeclientset  *kubernetes.Clientset
+	informer       coreinformers.NodeInformer
+	informerSynced cache.InformerSynced
+	lister         corelisters.NodeLister
+	recorder       record.EventRecorder
+	queue          workqueue.RateLimitingInterface
+}
+
+func NewNodeController(kubeclientset *kubernetes.Clientset, nodeInformer coreinformers.NodeInformer) (*NodeController, error) {
+	informer := nodeInformer
+	eventBroadcaster := record.NewBroadcaster()
+	recorder := eventBroadcaster.NewRecorder(scheme.Scheme, v1.EventSource{Component: "mizar-node-controller"})
+	eventBroadcaster.StartLogging(klog.Infof)
+	eventBroadcaster.StartRecordingToSink(
+		&v1core.EventSinkImpl{Interface: kubeclientset.CoreV1().Events("")})
+	queue := workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter())
+	c := &NodeController{
+		kubeclientset:  kubeclientset,
+		informer:       informer,
+		informerSynced: informer.Informer().HasSynced,
+		lister:         informer.Lister(),
+		recorder:       recorder,
+		queue:          queue,
+	}
+	klog.Infof(c.logInfoMessage("Sending events to api server"))
+	informer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(object interface{}) {
+			resource := object.(*v1.Node)
+			key := c.genKey(resource)
+			c.Enqueue(NODE_CREATE, key)
+			fmt.Printf("NODE_CREATE: %s", key)
+		},
+		UpdateFunc: func(oldObject, newObject interface{}) {
+			oldResource := oldObject.(*v1.Node)
+			newResource := newObject.(*v1.Node)
+			oldKey := c.genKey(oldResource)
+			newKey := c.genKey(newResource)
+			if oldKey == newKey {
+				klog.Infof(c.logEventMessage("No actual change in nodes, discarding: ", newKey))
+			} else {
+				command, err := c.compareKey(oldKey, newKey)
+				if err != nil {
+					klog.Errorf(c.logInfoMessage("Unexpected string in queue; discarding: " + newKey))
+				} else {
+					c.Enqueue(command, newKey)
+				}
+				fmt.Printf("NODE_UPDATE: %s, %s", command, newKey)
```

Comment:
consider to use klog to do the logging

Reply:
All Printf () are converted to klog.infof() or klog.errorf().